examples/api-gateway-lambda.hs

```diff
@@ -0,0 +1,199 @@
+{-# LANGUAGE OverloadedLists   #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+import           Control.Lens
+import           Data.Aeson                 (Value (Array), object)
+import qualified Data.ByteString.Lazy.Char8 as B
+import           Data.Text                  (Text)
+import           Stratosphere
+
+
+-- to curl the endpoint: (substitute your APIGW deployment URL)
+-- curl -v -X POST -H "Content-Type: application/json" -d "{\"description\": \"test task\"}" "https://t04rmwbtgj.execute-api.us-east-1.amazonaws.com/v1/stack"
+
+main :: IO ()
+main = B.putStrLn $ encodeTemplate myTemplate
+
+myTemplate :: Template
+myTemplate = template
+  [ role
+  , incomingS3Bucket
+  , lambda
+  , permission
+  , apiGWRestApi
+  , apiGWResource
+  , apiGWMethod
+  , apiGWDeployment
+  ]
+  & description ?~ "Simple api gateway attached to a lambda that writes the request body to a S3 object"
+  & formatVersion ?~ "2010-09-09"
+
+
+apiGWRestApi :: Resource
+apiGWRestApi = (resource "ApiGWRestApi" $
+  ApiGatewayRestApiProperties $
+  apiGatewayRestApi
+    "myApi"
+  )
+
+apiGWResource :: Resource
+apiGWResource = (resource "ApiGWResource" $
+  ApiGatewayResourceProperties $
+  apiGatewayResource
+    (GetAtt "ApiGWRestApi" "RootResourceId")
+    "stack"
+    (toRef apiGWRestApi)
+  )
+  & dependsOn ?~ deps [ apiGWRestApi ]
+
+apiGWMethod :: Resource
+apiGWMethod = (resource "ApiGWMethod" $
+  ApiGatewayVerbProperties $
+  apiGatewayVerb
+    "NONE"
+    "POST"
+    (toRef apiGWResource)
+    (toRef apiGWRestApi)
+    & agvIntegration ?~ integration
+    & agvMethodResponses ?~ [ methodResponse ]
+  )
+  & dependsOn ?~ deps [
+      apiGWRestApi
+    , apiGWResource
+    , lambda
+    ]
+
+  where
+    integration = apiGatewayIntegration "AWS"
+      & agiIntegrationHttpMethod ?~ "POST"
+      & agiUri ?~ (Join "" [
+          "arn:aws:apigateway:"
+        , Ref "AWS::Region"
+        , ":lambda:path/2015-03-31/functions/"
+        , GetAtt "WriteS3ObjectLambda" "Arn"
+        , "/invocations"])
+      & agiIntegrationResponses ?~ [ integrationResponse ]
+      & agiPassthroughBehavior ?~ "NEVER"
+      & agiRequestTemplates ?~ [ ("application/json", "{\"body\": $input.body}") ]
+
+    integrationResponse = apiGatewayIntegrationResponse
+      & agirResponseTemplates ?~ [ ("application/json", "$input.json('$.body')") ]
+      & agirStatusCode ?~ "200"
+
+    methodResponse = apiGatewayMethodResponse "200"
+
```

Comment:
Can you remove a lot of the extra newlines between functions? There's also a few newlines between list items as well for some reason.